flambda-backend/testsuite/tools/dune

```diff
@@ -13,7 +13,7 @@
 
 (executable
  (name codegen_main)
- (modes byte)
+ (modes native)
```

Comment:
Note: this unrelated change is here because one of the asmgen test failed, and the `codegen` binary used in this test proved completely useless for debugging. Using `native` mode means that the generated executable contains enough debug info to actually print backtraces, and can be debugged with a native debugger.